frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseCheckoutAction.java

```diff
@@ -0,0 +1,132 @@
+package com.virtuslab.gitmachete.frontend.actions.base;
+
+import static com.virtuslab.gitmachete.frontend.resourcebundles.GitMacheteBundle.getNonHtmlString;
+import static com.virtuslab.gitmachete.frontend.resourcebundles.GitMacheteBundle.getString;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.stream.Stream;
+
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.intellij.openapi.progress.ProgressIndicator;
+import com.intellij.openapi.progress.Task;
+import com.intellij.openapi.project.Project;
+import git4idea.branch.GitBranchUiHandlerImpl;
+import git4idea.branch.GitBranchWorker;
+import git4idea.commands.Git;
+import git4idea.repo.GitRepository;
+import io.vavr.control.Option;
+import kr.pe.kwonnam.slf4jlambda.LambdaLogger;
+import lombok.CustomLog;
+import lombok.experimental.ExtensionMethod;
+import lombok.val;
+import org.checkerframework.checker.guieffect.qual.UIEffect;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+import com.virtuslab.gitmachete.frontend.actions.expectedkeys.IExpectsKeySelectedBranchName;
+import com.virtuslab.gitmachete.frontend.resourcebundles.GitMacheteBundle;
+import com.virtuslab.gitmachete.frontend.vfsutils.GitVfsUtils;
+import com.virtuslab.qual.guieffect.UIThreadUnsafe;
+
+@ExtensionMethod({GitVfsUtils.class, GitMacheteBundle.class})
+@CustomLog
+public abstract class BaseCheckoutAction extends BaseGitMacheteRepositoryReadyAction
+    implements
+      IExpectsKeySelectedBranchName {
+
+  @Override
+  public LambdaLogger log() {
+    return LOG;
+  }
+
+  protected abstract @Nullable String getTargetBranchName(AnActionEvent anActionEvent);
+
+  protected abstract String getNonExistingBranchMessage(AnActionEvent anActionEvent);
```

Comment:
@PawelLipski the result of this is identified as `@Tainted`

Reply:
I think it might be due to the `.format` method or due to method call.

Reply:
Okay... and then the value is passed is to a place where an `@Untainted` value is expected?
In our interpretation:
* `@Tainted` = maybe has HTML
* `@Untainted` = surely does **not** have HTML

Reply:
> the value is passed is to a place where an @Untainted value is expected?

Yes, the `presentation.setDescription(value)`

Now, all the current implementations of the method only return `nonHtmlString`

Reply:
Now as a remedy, pls mark the return type of `getNonExistingBranchMessage` as `@Untainted` (non-HTML) as well (might be needed in both base class, and in concrete impls).

Post the compiler (Checker) errors in case of problems.

Reply:
Yes, and I tried adding `@Untainted` tag to the return type of `getNonExistingBranchMessage` and all the implementations, to further assist the compile, process. But still, the same problem.

Reply:
Note there is a short mention of _`@Tainted` and `@Untainted` annotations_ in CONTRIBUTING.md. If you think there's anything you can add there which might be useful for the future devs, feel free to include such change in this PR

Reply:
> Yes, and I tried adding @Untainted tag to the return type of getNonExistingBranchMessage and all the implementations, to further assist the compile, process. But still, the same problem.

Probably wrong import (from `javax`). Lmk if the problem persists